src/prefect/utilities/executors.py

```diff
@@ -57,17 +60,42 @@ def looper() -> None:
                 iters = (iters + 1) % self.interval
                 time.sleep(self.rate)
 
-        self.executor = ThreadPoolExecutor(max_workers=2)
+        def monitor() -> None:
+            while not self._exit:
+                if not self.fut.running():
```

Comment:
Is there a possibility where this can be called before `self.fut` it initialized?

Reply:
Good question - there is not, because I only submit the `monitor` function for execution _after_ I've created `self.fut` (see the lines below)

Reply:
Yeah I saw, just wanted to make sure 👍